src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -210,7 +211,13 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
       }
 
       logPoolState("Timeout failure ");
-      throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime), lastConnectionFailure.getAndSet(null)));
+      final Throwable cause = lastConnectionFailure.getAndSet(null);
+      final SQLTimeoutException exception = new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime)));
+
+      if (cause != null) {
```

Comment:
If you set cause to `null` then it is impossible to use `initCause()` to set it to something useful (if needed) because it is like already set. Not much improvement, I agree. Other thing that is fixed here is that current code passes cause as a second argument to `String.format()` instead of the second argument to exception constructor (i.e. misplaced bracket).

```java
- throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime), lastConnectionFailure.getAndSet(null)));
+ throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedTimeMs(startTime)), lastConnectionFailure.getAndSet(null));
```